uniffi_bindgen/src/bindings/kotlin/templates/RustBufferHelpers.kt

```diff
@@ -114,199 +261,133 @@ fun String.Companion.lift(rbuf: RustBuffer.ByValue): String {
 }
 
 fun String.Companion.read(buf: ByteBuffer): String {
-    val len = Int.read(buf)
+    val len = buf.getInt()
     val byteArr = ByteArray(len)
     buf.get(byteArr)
     return byteArr.toString(Charsets.UTF_8)
 }
 
-
-// I can't figure out how to make a generic implementation of (Any?).read, and IIUC there are some
-// restrictions on generics in Kotlin (inherited from the JVM) that make it impossible to write in the
-// style I want here. So, we use a standalone helper.
-
-fun<T> liftOptional(rbuf: RustBuffer.ByValue, readItem: (ByteBuffer) -> T): T? {
-    return liftFromRustBuffer(rbuf) { buf -> readOptional(buf, readItem) }
+fun String.lower(): RustBuffer.ByValue {
+    val byteArr = this.toByteArray(Charsets.UTF_8)
+    // Ideally we'd pass these bytes to `ffi_bytebuffer_from_bytes`, but doing so would require us
+    // to copy them into a JNA `Memory`. So we might as well directly copy them into a `RustBuffer`.
+    val rbuf = RustBuffer.alloc(byteArr.size)
+    rbuf.asByteBuffer()!!.put(byteArr)
+    return rbuf
 }
 
-fun<T> readOptional(buf: ByteBuffer, readItem: (ByteBuffer) -> T): T? {
-    if (! Boolean.read(buf)) {
-        return null
-    }
-    return readItem(buf)
+fun String.write(buf: RustBufferBuilder) {
+    val byteArr = this.toByteArray(Charsets.UTF_8)
+    buf.putInt(byteArr.size)
+    buf.put(byteArr)
 }
 
-fun<T> liftSequence(rbuf: RustBuffer.ByValue, readItem: (ByteBuffer) -> T): List<T> {
-    return liftFromRustBuffer(rbuf) { buf -> readSequence(buf, readItem) }
-}
+{% when Type::Optional with (inner_type) -%}
```

Comment:
Nit: we're getting to library specific code now.

Suggest adding a comment in the generate code to make this easier to find?